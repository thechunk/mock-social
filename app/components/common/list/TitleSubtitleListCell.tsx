import React, {PureComponent} from 'react';
import {Text, View} from 'react-native';
import ListCell from './ListCell';
import {default as inherited} from "./styles";
import g from '../../../styles/global';

class TitleSubtitleListCell extends PureComponent<ITitleSubtitleListCell> {
    render() {
        return (
            <ListCell
                onPress={this.props.onPress}
                containerStyle={inherited.cellOuter}
                cellStyle={this.props.cellStyle}>
                {this.props.leftAccessory
                    ? <View style={inherited.leftAccessory}>{this.props.leftAccessory()}</View>
                    : null}
                <View style={[g.flexColumn, {justifyContent: 'center'}]}>
                    <Text style={this.props.titleStyle}>{this.props.title}</Text>
                    {this.props.subtitle && this.props.subtitle.length > 0
                        ? <Text style={this.props.subtitleStyle}>{this.props.subtitle}</Text>
                        : null
                    }
                </View>
                {this.props.rightAccessory
                    ? <View style={inherited.rightAccessory}>{this.props.rightAccessory()}</View>
                    : null}
            </ListCell>
        );
    }
}

export default TitleSubtitleListCell;
