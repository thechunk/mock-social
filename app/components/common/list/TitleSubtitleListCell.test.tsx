import * as React from 'react';

import {shallow} from 'enzyme';
import TitleSubtitleListCell from './TitleSubtitleListCell';

describe('TitleSubtitleListCell', () => {
    it('renders correctly', () => {
        expect(shallow(<TitleSubtitleListCell key={1} title='title' />)).toMatchSnapshot();
    });
});
