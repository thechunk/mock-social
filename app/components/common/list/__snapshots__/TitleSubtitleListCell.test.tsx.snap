// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TitleSubtitleListCell renders correctly 1`] = `
<ListCell
  containerStyle={
    Object {
      "borderBottomColor": "rgba(0, 0, 0, 0.2)",
      "borderBottomWidth": 0.5,
    }
  }
>
  <View
    style={
      Array [
        Object {
          "flex": 1,
          "flexDirection": "column",
        },
        Object {
          "justifyContent": "center",
        },
      ]
    }
  >
    <Text>
      title
    </Text>
  </View>
</ListCell>
`;
